x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -224,81 +229,65 @@ public TokenService(Settings settings, Clock clock, Client client,
         getTokenMetaData();
     }
 
-    public static Boolean isTokenServiceEnabled(Settings settings) {
-        return XPackSettings.TOKEN_SERVICE_ENABLED_SETTING.get(settings);
-    }
-
     /**
      * Creates an access token and optionally a refresh token as well, based on the provided authentication and metadata with an
      * auto-generated token document id. The created tokens are stored in the security index.
      */
     public void createOAuth2Tokens(Authentication authentication, Authentication originatingClientAuth,
                                    Map<String, Object> metadata, boolean includeRefreshToken,
                                    ActionListener<Tuple<UserToken, String>> listener) {
-        createOAuth2Tokens(UUIDs.randomBase64UUID(), authentication, originatingClientAuth, metadata, includeRefreshToken, listener);
+        // the created token is compatible with the oldest node version in the cluster
+        final Version tokenVersion = getTokenVersionCompatibility();
+        // tokens moved to a separate index in newer versions
+        final SecurityIndexManager tokensIndex = getTokensIndexManagerForVersion(tokenVersion);
```

Comment:
This would be inconsistent with how we name `SecurityIndexManager securityIndex`.
Also, see https://github.com/elastic/elasticsearch/pull/30526#discussion_r188403979 .

Reply:
> This would be inconsistent with how we name SecurityIndexManager securityIndex.

Yeah, my comment was more about whether this should be called `XIndex` vs`XIndexManager`. I see the point but in the same line of thinking, shouldn't we be naming `getTokensIndexManagerForVersion` -> `getTokensIndexForVersion` ?  Now that I've read the code once and we discussed about it, it doesn't make such a significant difference, but thinking what would be more obvious/readable for someone looking into it. I don't have such a strong opinion tbh :)

Reply:
Okay, I have renamed `getTokensIndexManagerForVersion` to `getTokensIndexForVersion` .